Automattic/jetpack
# Connection: prevent potential version compatibility issues with the Licensing package

## Proposed changes:
PR #28509 introduced a [change](https://github.com/Automattic/jetpack/pull/28509/files#diff-3a10f158f2e15a9ce06885a9e4aa3fb86149a22d974a5ba6759539ba935537e3R62-R64) that might cause compatibility issues if the site runs the latest Connection package and an outdated Licensing package.

This PR introduces additional check to make sure the method is there before running it.

### Other information:


## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
Just making sure the existing functionality still works:
1. Use a site with no plans activated.
1. Go to https://cloud.jetpack.com/features/comparison and purchase the Jetpack Backup license. Do not attach it to a site.
2. Disconnect your site if it's connected.
3. Go to Jetpack Backup dashboard and click "Get VaultPress Backup". That should initiate the connection flow.
4. Click "Approve", you should get redirected to "My Jetpack -> Add a License Key" page.
5. Confirm that you see your unattached license in the dropdown.
